Modules/System/Azure Blob Services API/src/Helper/ABSOptionalParameters.Codeunit.al

```diff
@@ -226,6 +226,66 @@ codeunit 9047 "ABS Optional Parameters"
         SetRequestHeader('x-ms-requires-sync', ValueText);
     end;
 
+    /// <summary>
+    /// Sets the value for 'x-ms-lease-action' HttpHeader for a request.
+    /// </summary>
+    /// <param name="Value">Enum "ABS Lease Action" value specifying the HttpHeader value</param>    
+    internal procedure LeaseAction("Value": Enum "ABS Lease Action")
+    begin
+        SetRequestHeader('x-ms-lease-action', Format("Value"));
+    end;
+
+    /// <summary>
+    /// Sets the value for 'x-ms-lease-break-period' HttpHeader for a request.
+    /// </summary>
+    /// <param name="Value">Integer value specifying the HttpHeader value.</param>
+    procedure LeaseBreakPeriod("Value": Integer)
```

Comment:
Since this optional parameter is conditional on the lease action being `LeaseAction::Break`, I believe it's cleaner  to remove this procedure from optional parameters codeunit and take it as an argument in the API instead. This will also make it more transparent to the consumer of the API.

I know this will add a couple of new API overloads (with the different permutations of arguments) but I still think it's better than an optional parameter that is tied up to a specific lease action, and errors out in all other cases.

Reply:
Done